app/main/views/suppliers.py

```diff
@@ -325,108 +363,66 @@ def submit_company_summary():
-            abort(503)
-    else:
-        return render_template(
-            "suppliers/company_summary.html",
-            missing_fields=missing_fields,
-            **template_data
-        ), 400
-
 
-@main.route('/create-your-account', methods=['GET'])
-def create_your_account():
-    current_app.logger.info(
-        "suppliercreate: get create-your-account supplier_id:{}".format(
-            session.get('email_supplier_id', 'unknown')))
-    form = EmailAddressForm()
-
-    template_data = main.config['BASE_TEMPLATE_DATA']
-
-    return render_template(
-        "suppliers/create_your_account.html",
-        form=form,
-        **template_data
-    ), 200
-
-
-@main.route('/create-your-account', methods=['POST'])
-def submit_create_your_account():
-    current_app.logger.info(
-        "suppliercreate: post create-your-account supplier_id:{}".format(
-            session.get('email_supplier_id', 'unknown')))
-    template_data = main.config['BASE_TEMPLATE_DATA']
-    form = EmailAddressForm()
-
-    required_fields = [
-        "email_supplier_id",
-        "email_company_name",
-    ]
-
-    missing_fields = [field for field in required_fields if field not in session]
-
-    if missing_fields:
-        current_app.logger.info("Failed to create user as broken session", session)
-        abort(400)
-
-    if form.validate_on_submit():
-        token = generate_token(
-            {
-                "email_address":  form.email_address.data,
-                "supplier_id": session['email_supplier_id'],
-                "supplier_name": session['email_company_name']
-            },
-            current_app.config['SHARED_EMAIL_KEY'],
-            current_app.config['INVITE_EMAIL_SALT']
-        )
+            token = generate_token(
+                {
+                    "email_address":  account_email_address,
+                    "supplier_id": session['email_supplier_id'],
+                    "supplier_name": session['email_company_name']
+                },
+                current_app.config['SHARED_EMAIL_KEY'],
+                current_app.config['INVITE_EMAIL_SALT']
+            )
 
-        url = url_for('main.create_user', encoded_token=token, _external=True)
+            url = url_for('main.create_user', encoded_token=token, _external=True)
 
-        email_body = render_template(
-            "emails/create_user_email.html",
-            company_name=session['email_company_name'],
-            url=url
-        )
-        try:
-            send_email(
-                form.email_address.data,
-                email_body,
-                current_app.config['DM_MANDRILL_API_KEY'],
-                current_app.config['CREATE_USER_SUBJECT'],
-                current_app.config['RESET_PASSWORD_EMAIL_FROM'],
-                current_app.config['RESET_PASSWORD_EMAIL_NAME'],
-                ["user-creation"]
+            email_body = render_template(
+                "emails/create_user_email.html",
```

Comment:
I've just noticed that this template doesn't wrap the URL in an anchor tag at all.  I guess *most* mail clients will turn it into a link, but we might want to add it while we're fixing this up?